fsspec_reference_maker/combine.py

```diff
@@ -246,16 +246,25 @@ def _build_output(self, ds, ds0, fss):
 
     def _determine_dims(self):
         logger.debug("open mappers")
-        with fsspec.open_files(self.path, **self.storage_options) as ofs:
-            fss = [
-                fsspec.filesystem(
-                    "reference", fo=json.load(of),
-                    remote_protocol=self.remote_protocol,
-                    remote_options=self.remote_options
-                ) for of in ofs
-            ]
-            self.fs = fss[0].fs
-            mappers = [fs.get_mapper("") for fs in fss]
+
+        # If self.path is a list of dictionaries, pass them directly to fsspec.filesystem
+        if type(self.path[0]) == dict:
```

Comment:
`isinstance(self.path[0], dict)` - allows for subclasses of dict. `collections.abc.Mapping` may be even better. Also, the path may be a single string with a "*" character (like in `example_ensemble`!), so you would get a TypeError here.

Reply:
I think even if `path="*"` the code would work, since `self.path[0]` would return the `*` character as type `str`. I'll update it to use `isinstance` since we want to check for subclasses.
```suggestion
        import collections.abc
        if isinstance(self.path[0], collections.abc.Mapping):
```

Reply:
You can push this - I can't commit for some reason.

Reply:
Okay it should be pushed now